src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundUserAcceptanceTest.java

```diff
@@ -77,16 +78,25 @@ void testEdifactIsSuccessfullyProcessedAndPushedToGpOutboundQueue(String testGro
 
         getLabResultsMeshClient().sendEdifactMessage(outboundMeshMessage);
 
-        for (String expectedMessageBody : testData.getJsonList()) {
-            Message gpOutboundQueueMessage = getGpOutboundQueueMessage();
-            assertThat(gpOutboundQueueMessage).isNotNull();
-
-            String correlationId = gpOutboundQueueMessage.getStringProperty(JmsHeaders.CORRELATION_ID);
-            assertThat(correlationId).isNotEmpty();
-
-            String messageBody = parseTextMessage(gpOutboundQueueMessage);
-            assertFhirEquals(expectedMessageBody, messageBody);
-        }
+        var assertions = testData.getJsonList().stream()
+            .map(expectedMessageBody -> {
+                Message gpOutboundQueueMessage = getGpOutboundQueueMessage();
+                return Pair.of(expectedMessageBody, gpOutboundQueueMessage);
+            })
```

Comment:
Seems like we could just do `var jmsMessage = getGpOutboundQueueMessage()` in the second `map`, which would allow us to get rid of this transformation. Just a suggestion.